### PR TITLE
feat(action menu): add component tokens

### PR DESCRIPTION
**Pull request by alisonailea**
**Related Issue:** #7180 

## Summary

Adds Action-Menu component tokens and sub-component tokens to Calcite Components

### Action Group
--calcite-action-group-action-menu-border-color: The border color of the sub-component.
--calcite-action-group-action-menu-text-color: The text color of the sub-component.

### Action Menu
--calcite-action-menu-border-color: The border color of the component.
--calcite-action-menu-text-color: The text color of the component.

### Block

--calcite-block-action-menu-border-color: The border color of the sub-component.
--calcite-block-action-menu-text-color: The text color of the sub-component.

### Menu Item
--calcite-menu-item-action-menu-border-color: The border color of the sub-component.
--calcite-menu-item-action-menu-text-color: The text color of the sub-component.

### Panel
--calcite-panel-action-menu-border-color: The border color of the sub-component.
--calcite-panel-action-menu-text-color: The text color of the sub-component.

### CODE REVIEW

File: packages/calcite-components/src/components/action-menu/action-menu.scss
Comment (driskull) [inline packages/calcite-components/src/components/action-menu/action-menu.scss:51]: Should these be added as part of this PR? The popover vars already exist.

**Review by driskull — COMMENTED** [review]
This looks good besides the missing popover tokens.

Because the PR and commit reference just a feature for action-menu, i wouldn't expect tokens added to block, menu-item, panel, action-group. I think this is fine its just a little odd to be updating other components. I would expect the updates to the other components to handle the vars for action-menu once its merged.

File: packages/calcite-components/src/components/action-menu/action-menu.scss
Comment (alisonailea) [reply packages/calcite-components/src/components/action-menu/action-menu.scss:51]: Popover still needs to be refactored. It's next on my list. I was going to take care of it then.

**Comment by alisonailea** [comment]
> This looks good besides the missing popover tokens.
> 
> Because the PR and commit reference just a feature for action-menu, i wouldn't expect tokens added to block, menu-item, panel, action-group. I think this is fine its just a little odd to be updating other components. I would expect the updates to the other components to handle the vars for action-menu once its merged.

As stated in my suggested workflow proposal document. It should be up to the person with the most knowledge of those component tokens to update them in dependent components. This workflow means no component is blocked by any other.

**Review by driskull — APPROVED** [review]
This looks good. Some of the CSS files could use sorting of props but not a showstopper. 👍

**Pull request merged**
